File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        boolean flag = false;
        int ans = 0;
        int dup = x;
        if(x<0){
            flag = true;
            
        }
        while(x!=0){
            int temp = x%10;
            ans = ans*10 + temp;
            x=x/10;

        }

        if(flag == true){
        ans = ans * -1;
            }

        if(ans == dup) return true;
      
        return false;
        
    }
}

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        dup=x
        if x<0:
            return False
        
        value = 0
        while x>0:
            temp = x%10
            value = value*10+temp
            x=x//10
        return value==dup
